lib/main.js

```diff
@@ -218,20 +222,59 @@ export default {
           const col = Number.parseInt(match[2], 10) || 0;
           const isErr = (match[4] === 'E' && !this.pycodestyleErrorsToWarnings)
             || (match[4] === 'F' && this.flakeErrors);
-          const range = extractRange({
-            code: match[3],
-            message: match[5],
-            lineNumber: line,
-            colNumber: col,
-            textEditor,
-          });
-
-          messages.push({
-            type: isErr ? 'Error' : 'Warning',
-            text: `${match[3]} — ${match[5]}`,
-            filePath,
-            range,
-          });
+
+          try {
+            const range = extractRange({
+              code: match[3],
+              message: match[5],
+              lineNumber: line,
+              colNumber: col,
+              textEditor,
+            });
+            messages.push({
+              type: isErr ? 'Error' : 'Warning',
+              text: `${match[3]} — ${match[5]}`,
+              filePath,
+              range,
+            });
+          } catch (e) {
+            const flake8Version = await getVersionString(execPath);
+            const issueURL = 'https://github.com/AtomLinter/linter-flake8/issues/new';
+            const title = encodeURIComponent(`Flake8 rule '${match[3]}' reported an invalid point`);
+            const body = encodeURIComponent([
```

Comment:
One function does too much. Can you abstract out all the trace creation logic to a helper?